### PR TITLE
🚸 Hide `uninstall` target when it exists in superproject

### PR DESCRIPTION
**Pull request by ystade**
## Description

When QMap is included in another project that also provides an uninstall target, this leads to a conflict. This resolves this issue by creating this target only when there does not exist already one and fixes some bugs with the Z3 import.

## Checklist:

- [x] The pull request only contains commits that are related to it.
- [x] I have added appropriate tests and documentation.
- [x] I have made sure that all CI jobs on GitHub pass.
- [x] The pull request introduces no new warnings and follows the project's style guidelines.

### CODE REVIEW

File: src/CMakeLists.txt
Comment (burgholzer) [inline src/CMakeLists.txt:1]: Is this really the best way to go forward here? I think by now we simply require Z3 to be installed.

**Review by burgholzer — APPROVED** [review]
LGTM in principle. Thanks. Just an inline question on the Z3 related changes.

File: src/CMakeLists.txt
Comment (ystade) [reply src/CMakeLists.txt:1]: How you wish, I do not have any strong opinion here. However, then one should definitely also adapt the lines
https://github.com/cda-tum/mqt-qmap/blob/94ee532c53beb5b33823445d752b0356956d629f/cmake/ExternalDependencies.cmake#L8-L11

Comment (burgholzer) [reply src/CMakeLists.txt:1]: Nah, nevermind. But could you please just move the alias definitions for the exact mapper and the Clifford synthesis library into the if block just a couple lines above the newly created one? Makes much more sense to me to have these right after the corresponding add_qmap_library calls anyways.

Comment (ystade) [reply src/CMakeLists.txt:1]: Done ✅

**Review by burgholzer — APPROVED** [review]
Many thanks.
The test failures are due to Qiskit 1.1 being released overnight and introducing new warning.
The respective warning here should be ignored in our pyproject.toml
Do you know how to do that or should I give it a try later on?

**Comment by ystade** [comment]
> Many thanks. The test failures are due to Qiskit 1.1 being released overnight and introducing new warning. The respective warning here should be ignored in our pyproject.toml Do you know how to do that or should I give it a try later on?

No, I have no clue how... :/

**Comment by burgholzer** [comment]
> > Many thanks. The test failures are due to Qiskit 1.1 being released overnight and introducing new warning. The respective warning here should be ignored in our pyproject.toml Do you know how to do that or should I give it a try later on?
> 
> No, I have no clue how... :/

https://github.com/cda-tum/mqt-qmap/pull/454/commits/70c854854def3daf4263db3ceb05436a126573ad should do the trick.
The syntax for this is available under the following link if you should ever need it: https://docs.python.org/3/library/warnings.html#describing-warning-filters

**Pull request merged**